harness/suite.go

```diff
@@ -203,6 +211,8 @@ func (s *Suite) Run() (err error) {
 		return err
 	}
 
+	defer s.reporters.Output(s.outputPath)
```

Comment:
I'm still not a fan of passing in the "outputPath" function for the reporter to call vs passing in the `s.opts.OutputDir` directory here and having the reporter call it vs, say, doing 

```
reportDir := s.outputPath("reports")
os.Mkdir(reportDir)
...
reporters.Output(reportDir)
```

and then having each reporter append its own filename to that given directory.

It feels weird to leak a private function into the reporters abstraction when all that function does is a `filepath.Join`.